src/App.js

```diff
@@ -1,19 +1,40 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
-  return (
-    <div id="App">
-      <header>
-        <h1>Application title</h1>
-      </header>
-      <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
-      </main>
-    </div>
-  );
-};
+	const [messages, setMessages] = useState(chatMessages);
 
+	const totalLikes = messages.reduce((total, entry) => {
+		total += entry.liked;
+		return total;
+	}, 0);
+	const setLike = (id) => {
+		setMessages((prev) => {
+			return prev.map((entry) => {
+				if (id === entry.id) {
+					return { ...entry, liked: !entry.liked };
+				} else {
+					return entry;
+				}
+			});
+		});
+	};
+	return (
+		<div id='App'>
+			<header>
+				<h1>
+					Chat Log between {messages[0].sender} & {messages[1].sender}
```

Comment:
What if textMessage[0] and textMessage[1] were both messages from the same person? How else could you find a way to get the names of the senders?